### PR TITLE
feat(company name): update input pattern

### PR DESCRIPTION
**Pull request by manojava-gk**
## Description

update company name pattern and respective unit test cases. add companyName query param to the subscription url
update company name pattern in application requests section as we as invite business partner form

## Why

enhance search feature

## Issue

NA

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [ ] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [ ] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [ ] I have added tests that prove my changes work
- [ ] I have checked that new and existing tests pass locally with my changes
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by nidhigarg-bmw — APPROVED** [review]
lgtm

**Review by lavanya-bmw — CHANGES_REQUESTED** [review]
Please add change log

File: src/components/pages/Admin/components/RegistrationRequests/components/RequestList/index.tsx [hunk 1]
```diff
@@ -126,7 +127,7 @@ export const RequestList = ({
   }, [loaded])
 
   const onValidate = (expr: string) => {
-    const validateExpr = /^[ A-Za-z0-9]*$/.test(expr)
+    const validateExpr = Patterns.COMPANY_NAME.test(expr)
```
Comment (oyo) [inline src/components/pages/Admin/components/RegistrationRequests/components/RequestList/index.tsx:130]: please use from Patterns.ts:

`isCompanyName(expr)`

**Review by oyo — CHANGES_REQUESTED** [review]
one tiny change - otherwise okay

**Review by oyo — APPROVED** [review]

**Pull request merged**